mantidimaging/core/operations/loader.py

```diff
@@ -4,36 +4,40 @@
 import os
 import pkgutil
 import sys
-from typing import List, Protocol, cast, Union, TYPE_CHECKING
-from importlib.machinery import FileFinder, ModuleSpec
-from importlib.abc import Loader
+from importlib.util import module_from_spec
+from typing import List, TYPE_CHECKING
 
 if TYPE_CHECKING:
-    from PyInstaller.loader.pyimod02_importers import FrozenImporter
     from mantidimaging.core.operations.base_filter import BaseFilter
 
-MODULES_OPERATIONS: dict[str, Union['FrozenImporter', Loader]] = {}
-if not MODULES_OPERATIONS:
-    for finder, module_name, _ in pkgutil.walk_packages([os.path.dirname(__file__)]):
+_OPERATION_MODULES_LIST: List[BaseFilter] = []
+
+
+def _find_operation_modules() -> List[BaseFilter]:
+    module_list: List[BaseFilter] = []
+    for finder, module_name, ispkg in pkgutil.walk_packages([os.path.dirname(__file__)]):
+        if not ispkg:
+            continue
+
         if getattr(sys, 'frozen', False):
-            # If we're running a PyInstaller executable then we will get back a FrozenImporter object instead of a
-            # FileFinder. FrozenImporter implements load_module directly, but needs to use the full import name for
-            # the module to load it from the PYZ archive.
-            assert hasattr(finder, 'load_module')
-            MODULES_OPERATIONS[f'mantidimaging.core.operations.{module_name}'] = finder
-        else:
-            assert isinstance(finder, FileFinder)
-            spec = finder.find_spec(module_name)
-            assert isinstance(spec, ModuleSpec)
-            assert isinstance(spec.loader, Loader)
-            MODULES_OPERATIONS[module_name] = spec.loader
+            # If we're running a PyInstaller executable then we need to use a full module path
+            module_name = f'mantidimaging.core.operations.{module_name}'
 
+        # near impossible to type check as find can be a pyinstaller specific type that we can't normally import
+        spec = finder.find_spec(module_name)  # type: ignore[call-arg]
 
-class OperationModule(Protocol):
-    FILTER_CLASS: BaseFilter
+        assert spec is not None
```

Comment:
These are to keep `mypy` happy because `finder.find_spec` returns an `Optional`. I don't think there is any normal case where `walk_packages()` would give something that cause an issue `find_spec`, so if it did happen we'd want a full trace.